### PR TITLE
Possible to pass :scope_by option

### PR DESCRIPTION
**Pull request by emaxi**
Examples:
```
class Car < ActiveRecord::Base
    protokoll :code, :scoped_by => lambda { |o| "#{o.manufacturer}-#{o.year}" }
end
```
or

```
class Car < ActiveRecord::Base
    protokoll :code, :scope_by => :manufacturer
end
```

### CODE REVIEW

**Comment by emaxi** [comment]
Examples:
```
class Car < ActiveRecord::Base
    protokoll :code, :scoped_by => lambda { |o| "#{o.manufacturer}-#{o.year}" }
end
```
or

```
class Car < ActiveRecord::Base
    protokoll :code, :scope_by => :manufacturer
end
```

File: README.md [hunk 1]
```diff
@@ -81,6 +81,30 @@ end
 # will produce => "2011HOUSE00001", "2011HOUSE00002"...
 ```
 
+It's possible to pass :scope_by option as a simple method, Proc.new or lambda
+
+Ex:
+```ruby
+# :manufacturer should be a Car's instance method(like an ActiveRecord column)
+class Car < ActiveRecord::Base
+    protokoll :code, :scope_by => :manufacturer
```
Comment (celsodantas) [inline README.md]: I feel we should have a single name for this. `scope_by` should accept a lambda or a method name.
this way it simplifies the API.

thoughts?

File: lib/generators/protokoll/migration/templates/create_custom_auto_increments.rb [hunk 2]
```diff
@@ -2,11 +2,13 @@ class CreateCustomAutoIncrements < ActiveRecord::Migration
   def up
     create_table :custom_auto_increments, :force => true do |t|
       t.string  :counter_model_name
+      t.string  :counter_model_scope
```
Comment (celsodantas) [inline lib/generators/protokoll/migration/templates/create_custom_auto_increments.rb]: I wonder if the counter by scope should be an add on and have its own migration. That way only the people that actually need the scope would run a `rails g protokoll:migration:scope_by` and the scope migration would be created (or some equivalente generator).

I'm not a fan of pushing this out and having the people not using the scope feature with a blank column in the database.

File: README.md (hunk 1 above)
Comment (celsodantas) [inline README.md]: it seems that's what you did, you have a typo in the readme then =P

**Comment by celsodantas** [comment]
thanks for the PR man. =)  really appreciate it.

File: README.md (hunk 1 above)
Comment (emaxi) [inline README.md]: Yes, you are right, typo fixed.

**Comment by emaxi** [comment]
I implemented the generator as you suggest
`protokoll:migration:scope_by`

And I updated the README for how to enable scope_by feature.

Sorry for my long delay after your feedback.
Thanks

**Comment by anthonator** [comment]
Anything need to happen for this to be merged and released?

**Comment by emaxi** [comment]
I think is ready to merge, what do you think @celsodantas?

**Comment by celsodantas** [comment]
just need to work on the upgrade path for existing users.

need to update the major version and write some documentation. 
will do this weekend. Sorry for the delay.

**Comment by anthonator** [comment]
@celsodantas were you able to get to this over the weekend?

**Comment by celsodantas** [comment]
Sorry. Didn't have the time to do it. 

I'll merge this as is and work on the version bump on the weekend. In the meantime, if you need this feature, I suggest pointing bundler to the github merge commit.

**Pull request merged**